src/js/ripple/orderbook.js

```diff
@@ -1176,4 +1335,307 @@ OrderBook.prototype.is_valid = function() {
+      // Leg one out is greater than leg two in. This means we must clamp down
+      // on leg one out to have it match leg two in
+      var legOneQuality = this.getOfferQuality(legOneOffer);
+      var updatedTakerGets;
+
+      autobridgedTakerGets = this.getOfferTakerGetsFunded(legTwoOffer);
+      autobridgedTakerPays = legTwoTakerPaysFunded.multiply(legOneQuality);
+
+      if (legOneOffer.Account === legTwoOffer.Account) {
+        var legOneTakerGets = this.getOfferTakerGets(legOneOffer);
+        updatedTakerGets = legOneTakerGets.subtract(legTwoTakerPaysFunded);
+
+        this.setLegOneTakerGets(legOneOffer, updatedTakerGets);
+
+        this.clampLegOneOwnerFunds(legOneOffer);
+      } else {
+        // Update funded amount since leg one offer was not completely consumed
+        updatedTakerGets = legOneTakerGetsFunded
+          .subtract(legTwoTakerPaysFunded);
+
+        this.setLegOneTakerGetsFunded(legOneOffer, updatedTakerGets);
+      }
+
+      legTwoPointer++;
+    } else if (legTwoTakerPaysFunded.compareTo(legOneTakerGetsFunded) > 0) {
+      // Leg two in is greater than leg one out. This means we must clamp down
+      // on leg two in
+      var legTwoQuality = this.getOfferQuality(legTwoOffer);
+
+      autobridgedTakerGets = legOneTakerGetsFunded.divide(legTwoQuality);
+      autobridgedTakerPays = this.getOfferTakerPaysFunded(legOneOffer);
+
+      // Update funded amount since leg two offer was not completely consumed
+      legTwoOffer.taker_gets_funded = this.getOfferTakerGetsFunded(legTwoOffer)
+        .subtract(autobridgedTakerGets)
+        .to_text();
+      legTwoOffer.taker_pays_funded = legTwoTakerPaysFunded
+        .subtract(legOneTakerGetsFunded)
+        .to_text();
+
+      legOnePointer++;
+    } else {
+      // Rare case when leg one out and leg two in are equal. Neither leg needs
+      // to be clamped and both offers are fully consumed
+      autobridgedTakerGets = this.getOfferTakerGetsFunded(legTwoOffer);
+      autobridgedTakerPays = this.getOfferTakerPaysFunded(legOneOffer);
+
+      legOnePointer++;
+      legTwoPointer++;
+    }
+
+    var autobridgedOffer = this.formatAutobridgedOffer(
+      autobridgedTakerGets,
+      autobridgedTakerPays
+    );
+    autobridgedOffer.index = legOnePointer + legTwoPointer;
+
+    this._offersAutobridged.push(autobridgedOffer);
+  }
+};
+
+/**
+ * Format an autobridged offer and compute synthetic values (e.g. quality)
+ *
+ * @param [Amount] takerGets
+ * @param [Amount] takerPays
+ *
+ * @return [Object]
+ */
+
+OrderBook.prototype.formatAutobridgedOffer = function(takerGets, takerPays) {
+  assert(takerGets instanceof Amount, 'Autobridged taker gets is invalid');
+  assert(takerPays instanceof Amount, 'Autobridged taker pays is invalid');
+
+  var autobridgedOffer = {};
+  var quality = takerPays.divide(takerGets);
+
+  autobridgedOffer.TakerGets = {
+    value: takerGets.to_text(),
+    currency: this._currencyGets.to_hex(),
```

Comment:
We do this because to_json() will lose information, serialising as just an iso code? Maybe we should update to_json() to use to_hex() when it's needed for lossless encoding?